### PR TITLE
feat(deepinfra): embeddings

### PR DESCRIPTION
**Pull request by ovuruska**

### CODE REVIEW

File: packages/core/src/embeddings/DeepInfraEmbedding.ts [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+  input_tokens: number;
+  request_id: string;
+  inference_status: InferenceStatus;
+}
+
+export interface InferenceStatus {
+  status: string;
+  runtime_ms: number;
+  cost: number;
+  tokens_generated: number;
```
Comment (ichernev) [inline packages/core/src/embeddings/DeepInfraEmbedding.ts]: you can drop this

File: packages/core/src/embeddings/DeepInfraEmbedding.ts [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+const DEFAULT_MODEL = "sentence-transformers/clip-ViT-B-32";
+
+const API_TOKEN_ENV_VARIABLE_NAME = "DEEPINFRA_API_TOKEN";
+
+const API_ROOT = "https://api.deepinfra.com/v1/inference";
+
+export interface DeepInfraEmbeddingResponse {
+  embeddings: number[][];
+  input_tokens: number;
```
Comment (ichernev) [inline packages/core/src/embeddings/DeepInfraEmbedding.ts]: Currently there are 2 places where we return the number of input_tokens, it makes sense to leave only one. Pick based on what is done more commonly in other providers. Or put it in Status (it's metadata anyway).

File: packages/core/src/embeddings/DeepInfraEmbedding.ts [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+  }
+
+  async getQueryEmbeddings(queries: string[]): Promise<number[][]> {
+    const queriesWithPrefix = mapPrefixWithInputs(this.queryPrefix, queries);
+    return await this.getDeepInfraEmbedding(queriesWithPrefix);
+  }
+
+  private async getDeepInfraEmbedding(inputs: string[]): Promise<number[][]> {
+    const url = this.getUrl(this.model);
+    const response = await fetch(url, {
```
Comment (ichernev) [inline packages/core/src/embeddings/DeepInfraEmbedding.ts]: retry/timeout?

**Review by ichernev — APPROVED** [review]
Overall looks good, a few notes.

File: packages/core/src/embeddings/DeepInfraEmbedding.ts (hunk 2 above)
Comment (ovuruska) [reply packages/core/src/embeddings/DeepInfraEmbedding.ts]: Alright, so I'm leaving input_tokens in the Status object.

File: packages/core/src/embeddings/DeepInfraEmbedding.ts (hunk 1 above)
Comment (ichernev) [reply packages/core/src/embeddings/DeepInfraEmbedding.ts]: not droped :)

**Pull request closed without merging**